Update dev exp landing (#537)
canonical/launchpad-manual: docs/developer/explanation/ideas-behind-launchpad/index.rst

The ideas behind Launchpad
==========================

> Before writing a line of Launchpad code it helps to understand the foundational

Reviewer: ```suggestion
Before beginning to contribute to Launchpad, it helps to understand the foundational
```
